lightning/src/onion_message/messenger.rs

```diff
@@ -408,10 +408,16 @@ where
 	fn create_blinded_paths<
 		T: secp256k1::Signing + secp256k1::Verification
 	>(
-		&self, recipient: PublicKey, peers: Vec<ForwardNode>, secp_ctx: &Secp256k1<T>,
+		&self, recipient: PublicKey, peers: Vec<ForwardNode>, secp_ctx: &Secp256k1<T>, count: usize
 	) -> Result<Vec<BlindedPath>, ()> {
 		// Limit the number of blinded paths that are computed.
-		const MAX_PATHS: usize = 3;
+		const MAX_PATHS: usize = 10;
+
+		debug_assert!(count <= MAX_PATHS,
+			"Requested {} blinded_paths, but only a maximum of {} will be returned.",
+			count,
+			MAX_PATHS
+		);
```

Comment:
We probably don't need this if `ChannelManager` is passing how many paths it wants. It was only needed earlier because there was no way for `ChannelManager` to signal it.

Reply:
Makes sense!
Remove the redundant constant and check in [f32a93a](https://github.com/lightningdevkit/rust-lightning/commit/f32a93acc60d20078c42b96dbb4466969b06f5b3)
Thanks for the pointer!